### PR TITLE
API test for createSubAdmin

### PR DESCRIPTION
**Pull request by paurakhsharma**
## Description
API test for createSubAdmin
This PR has one skipped scenarios which will pass only after resolving issue #31276

## Related Issue
#31533 

## How Has This Been Tested?
Locally

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Acceptance test

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: tests/acceptance/features/apiProvisioning-v1/createSubAdmin.feature [hunk 1]
```diff
@@ -30,3 +32,16 @@ So that I can give administrative privilege of a group to a user
 			| groupid | not-group |
 		Then the OCS status code should be "102"
 		And the HTTP status code should be "200"
+
+	Scenario: subadmin of a group cannot make another user subadmin of their group
+		Given user "subadmin" has been created
+		And user "brand-new-user" has been created
+		And group "new-group" has been created
+		And user "subadmin" has been made a subadmin of group "new-group"
+		When user "admin" sends HTTP method "POST" to API endpoint "/cloud/users/brand-new-user/groups" with body
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v1/createSubAdmin.feature]: And user "admin" has sent HTTP method...

File: tests/acceptance/features/apiProvisioning-v2/createSubAdmin.feature [hunk 2]
```diff
@@ -30,3 +32,18 @@ So that I can give administrative privilege of a group to a user
...
 		Then the OCS status code should be "400"
 		And the HTTP status code should be "400"
+
+	@skip @issue-31276
+		Scenario: subadmin of a group cannot make another user subadmin of their group
+		Given user "subadmin" has been created
+		And user "brand-new-user" has been created
+		And group "new-group" has been created
+		And user "subadmin" has been made a subadmin of group "new-group"
+		When user "admin" sends HTTP method "POST" to API endpoint "/cloud/users/brand-new-user/groups" with body
```
Comment (phil-davis) [inline tests/acceptance/features/apiProvisioning-v2/createSubAdmin.feature]: same change here

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 3]
```diff
@@ -1132,6 +1132,50 @@ public function theAppsShouldInclude($appList) {
 		}
 	}
 
+	/**
+	 * @Then /^the user "([^"]*)" is the subadmin of the group "([^"]*)"$/
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Provisioning.php]: the user "abc" should be a subadmin of the group "xyz"

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 4]
```diff
@@ -1132,6 +1132,50 @@ public function theAppsShouldInclude($appList) {
...
+	 * @return void
+	 */
+	public function theUserIsTheSubadminOfTheGroup($user, $group) {
+		$fullUrl = $this->getBaseUrl() . "/ocs/v2.php/cloud/groups/$group/subadmins";
+		$client = new Client();
+		$options = [];
+		$options['auth'] = [$this->getAdminUsername(),$this->getAdminPassword()];
+		$this->response = $client->get($fullUrl, $options);
+		$listOfSubadmins = $this->getArrayOfSubadminsResponded($this->response);
+		print_r($listOfSubadmins);
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Provisioning.php]: remove debug code

File: tests/acceptance/features/bootstrap/Provisioning.php [hunk 5]
```diff
@@ -1132,6 +1132,50 @@ public function theAppsShouldInclude($appList) {
...
+		PHPUnit_Framework_Assert::assertContains(
+			$user,
+			$listOfSubadmins
+		);
+	}
+
+	/**
+	 * @Then /^the user "([^"]*)" is not the subadmin of the group "([^"]*)"$/
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/Provisioning.php]: Then the user "abc" should not be a subadmin of the gorup "xyz"

**Review by phil-davis — COMMENTED** [review]
see comments

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
silly drone network hungup error - drone has been restarted

**Comment by paurakhsharma** [comment]
Yes, thank you.
We need smarter drones :laughing:

**Comment by phil-davis** [comment]
Backport ``stable10`` #31332

**Pull request merged**